File: py/valid-sudoku/valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        valid = True
        for i in range(9):
            for j in range(9):
                if board[i][j] == ".":
                    continue
                if not isValid(board, i, j):
                    return False
        return valid

def isValid(board, row, col):
    b = board[row][col]
    for i in range(9):
        r = 3 * (row // 3) + i // 3
        c = 3 * (col // 3) + i % 3
        if (board[row][i] == b and col != i) or (board[i][col] == b and row != i) or (board[r][c] == b and row != r and col != c):
            return False
    return True

File: go/valid-sudoku/valid-sudoku.go
func isValidSudoku(board [][]byte) bool {
    valid := true
    for i:=0; i < 9; i++ {
        for j := 0; j < 9; j++ {
            if board[i][j] == byte('.') {
                continue
            }
            if !isValid(board, i, j){
                return false
            }
        } 
    }
    return valid
}

func isValid(board [][]byte, row int, col int) bool{
    b := board[row][col]
    for i:=0; i < 9; i++ {
        r := 3 * (row / 3) + i / 3
        c := 3 * ( col / 3) + i % 3
        if (board[row][i] == b && col != i) || (board[i][col] == b && row != i) || (board[r][c] == b && row != r && col != c) {
            
            return false
        }
    }
    return true
}
